benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/tasks/ready/ReadyTask.java

```diff
@@ -0,0 +1,140 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * @author Jesper Findahl (jesper.findahl@usi.ch)
+ *         created on 2017-04-20
+ */
+public class ReadyTask implements Runnable {
+
+    private static Logger logger = LoggerFactory.getLogger(ReadyTask.class.getSimpleName());
+
+    private final String testID;
+    private final String testDefinitionYamlString;
+    private final InputStream deploymentDescriptorInputStream;
+    private final Map<String, InputStream> bpmnModelInputStreams;
+
+    // services
+    private final MinioService minioService;
+    private final ExplorationModelDAO explorationModelDAO;
+
+    private final BenchFlowTestTaskController taskController;
+
+
+    public ReadyTask(String testID, String testDefinitionYamlString, InputStream deploymentDescriptorInputStream, Map<String, InputStream> bpmnModelInputStreams) {
+
+        this.testID = testID;
+        this.testDefinitionYamlString = testDefinitionYamlString;
+        this.deploymentDescriptorInputStream = deploymentDescriptorInputStream;
+        this.bpmnModelInputStreams = bpmnModelInputStreams;
+
+        this.taskController = BenchFlowTestManagerApplication.getTestTaskController();
+        this.minioService = BenchFlowTestManagerApplication.getMinioService();
+        this.explorationModelDAO = BenchFlowTestManagerApplication.getExplorationModelDAO();
+    }
+
+    @Override
+    public void run() {
+
+        logger.info("preparing test with ID " + testID);
+
+
+        // extract contents
+        InputStream definitionInputStream = IOUtils.toInputStream(testDefinitionYamlString, StandardCharsets.UTF_8);
+
+        // TODO - handle different SUT types
+        // TODO - check that termination criteria with time has not been exceeded
+
+        // save PT archive contents to Minio
+        minioService.saveTestDefinition(testID, definitionInputStream);
+        minioService.saveTestDeploymentDescriptor(testID, deploymentDescriptorInputStream);
+
+        bpmnModelInputStreams.forEach((fileName, inputStream) -> minioService.saveTestBPMNModel(testID,
+                fileName,
+                inputStream));
+
+        try {
+            BenchFlowTest test = BenchFlowDSL.testFromYaml(testDefinitionYamlString);
+
+            List<Integer> workloadUserSpace = generateExplorationSpace(test);
+
+            if (workloadUserSpace != null) {
+                explorationModelDAO.setWorkloadUserSpace(testID, workloadUserSpace);
+            }
+
+            setExperimentSelectionStrategy(test);
+
+            taskController.runDetermineExecuteExperimentsTask(testID);
+
+        } catch (BenchFlowDeserializationException | BenchFlowTestIDDoesNotExistException e) {
+            // should not happen since it has already been tested/added
+            logger.error("should not happen");
+            e.printStackTrace();
+        }
+
+
+    }
+
+    public static List<Integer> generateExplorationSpace(BenchFlowTest test) {
```

Comment:
Like discussed I put them here until we have a clearer picture of our needs. Probably we also need to make adjustments in the DSL.